### PR TITLE
Update build scripts

### PR DESCRIPTION
**Pull request by hall**
These changes fix a few runtime issues we were having with the image running on alpine but trying to use bash.

This also changes the Actions CI to build on every push, tag the repo if the version in `package.json` was changed, and then tag the image with all of the git tags at `HEAD`.

@ebauman, let me know what you think. In particular, I'm not sure if you want to create a git tag (and therefore image tag) _only_ when the version in `package.json` is bumped (which is what these changes are currently doing) or if you want to always tag the repo and image with whatever is in `package.json` on every build.

That sounds like a confusing explanation but what I'm trying to get at is do we want this check before updating the tags:

```bash
if git diff HEAD~1 package.json | grep '"version":'
```

### CODE REVIEW

**Comment by ebauman** [comment]
@hall I think I'd like to move this repo (and others) to branch-per-release model and have master be where dev lives. If that makes sense? It's how rancher/rancher does it, and what's nice is that you can then pull fixes into a particular release branch instead of having to strangely backtrack in master and re-tag.

So, what I'm thinking here is that - following rancher/rancher's lead - we begin issuing "release/x.x.x" branches that are always buildable for that release. i.e. they almost become the "master" of that release and should always be seen as buildable.

Now, given that knowledge, I'm thinking we should adopt adjusting `package.json` with the build tag instead of intrinsically tying the release version into the code itself. If that makes sense? 

What are your thoughts

**Comment by hall** [comment]
Yeah, that approach sounds good to me.

I'm not clear on what you mean by that last part though. You want `package.json` to get its version from the build tag?

**Comment by ebauman** [comment]
Yeah, that's what I was driving towards lol

**Comment by hall** [comment]
@ebauman, I've pushed some changes (https://github.com/hobbyfarm/ui/pull/57/commits/7236677db36b9f1c8af234802c3a3c456cc13200) that get the app version from either the git tag (if one is at `HEAD`) or the short hash of the current commit.

**Comment by ebauman** [comment]
Ok I will review this shortly and let you know. Thanks

File: .github/workflows/publish-image.yaml [hunk 1]
```diff
@@ -1,29 +1,18 @@
 name: publish image
-
-on:
-  push:
-    branches:
-      - master
-    paths:
-      - 'package.json'
-
+on: push
```
Comment (ebauman) [inline .github/workflows/publish-image.yaml]: Can we run this job only on pushes to a release branch, e.g. `release/1.2.3` ?

Comment (hall) [reply .github/workflows/publish-image.yaml]: Done :+1: we might want to consider building on PRs as well but I'm not sure at this point.

**Comment by ebauman** [comment]
LGTM

**Pull request merged**